### ISSUE TITLE
(custom_resources): retry after Rate exceeded

### ISSUE THREAD

**Issue by mattiamatrix** [aws/aws-cdk#24210]
### Describe the bug

In our project, we use AwsCustomResource to run many API calls against Lake Formation (`grantPermissions`, `revokePermissions`.)

However, in many cases, it happens that the deployment fails after a `Rate exceeded` error.

### Expected Behavior

It would be nice to have the opinion to enable an exponential backoff retry

### Current Behavior

The deployment fails with no other options.

### Reproduction Steps

Steps
```ts
new AwsCustomResource(this.scope, id, {
      role: this.customResourceRole,
      onCreate: {
        service: 'LakeFormation',
        action: 'grantPermissions',
        parameters: {
          Principal: { ... },
          Resource: {
            LFTagPolicy: { ... },
          Permissions: permissions,
        },
        physicalResourceId: PhysicalResourceId.of(id),
      },
```

### CDK CLI Version

2.65.0

### Node.js Version

18

### OS

macOs 13.0.1

### Language

Typescript

**Comment by pahud** [comment]
Yeah it would be great to have this feature. I am making it as p2 and any PR submission would be highly appreciated!

**Comment by kemalunel** [comment]
you have to implement retry behavior mechanism to your code. see https://docs.aws.amazon.com/sdkref/latest/guide/feature-retry-behavior.html
